kwasm-lemmas.md

```diff
@@ -321,52 +321,39 @@ Lemmas about `#getRange` and `#setRange` simplification:
     rule #getRange(_, _, WIDTH) <<Int SHIFT <Int MAX => true requires 2 ^Int ((8 *Int WIDTH) +Int SHIFT) <=Int MAX                          [simplification]
     rule VAL1 +Int VAL2                     <Int MAX => true requires VAL1 <Int MAX andBool VAL2 <Int MAX andBool #distinctBits(VAL1, VAL2) [simplification]
 
-    rule #getRange(BM, ADDR, WIDTH) >>Int SHIFT => #getRange(BM, ADDR +Int 1, WIDTH -Int 1) >>Int (SHIFT -Int 8)
-      requires 0 <=Int ADDR
-       andBool 0  <Int WIDTH
-       andBool 8 <=Int SHIFT
-       andBool #isByteMap(BM)
-      [simplification]
+    syntax Bool ::= #distinctBits ( Int , Int ) [function]
+ // ------------------------------------------------------
+    rule #distinctBits(#getRange(_, _, WIDTH), #getRange(_, _, _) <<Int SHIFT) => true requires WIDTH *Int 8 <=Int SHIFT
+```
 
-    rule #getRange(BM, ADDR, WIDTH) modInt MAX => #getRange(BM, ADDR, WIDTH -Int 1) modInt MAX
-      requires 0 <Int MAX andBool 0 <Int WIDTH
-       andBool 2 ^Int (8 *Int (WIDTH -Int 1)) modInt MAX ==Int 0
-       andBool #isByteMap(BM)
-      [simplification]
+`#wrap` over `#getRange`:
 
+```k
     rule #wrap(MAX_WIDTH, #getRange(BM, ADDR, WIDTH)) => #getRange(BM, ADDR, minInt(MAX_WIDTH, WIDTH))
       requires #isByteMap(BM)
       [simplification]
-
-    syntax Bool ::= #distinctBits ( Int , Int ) [function]
- // ------------------------------------------------------
-    rule #distinctBits(#getRange(_, _, WIDTH), #getRange(_, _, _) <<Int SHIFT) => true requires WIDTH *Int 8 <=Int SHIFT
 ```
 
-`#getRange` over `#setRange`
+Arithmetic over `#getRange`:
 
 ```k
-    rule #setRange(BM, ADDR, #getRange(BM, ADDR, WIDTH), WIDTH) => BM [simplification]
-    rule #setRange(BM, ADDR, (#getRange(_, _, WIDTH1) #as VAL1) +Int (VAL2 <<Int SHIFT), WIDTH)
-      => #setRange(#setRange(BM, ADDR, VAL1, minInt(WIDTH1, WIDTH)), ADDR +Int WIDTH1, VAL2, WIDTH -Int WIDTH1)
+    rule #getRange(BM, ADDR, WIDTH) >>Int SHIFT => #getRange(BM, ADDR +Int 1, WIDTH -Int 1) >>Int (SHIFT -Int 8)
       requires 0 <=Int ADDR
        andBool 0  <Int WIDTH
-       andBool WIDTH1 *Int 8 ==Int SHIFT
-      [simplification]
-
-    rule #getRange(#setRange(BM, EA, VALUE, SET_WIDTH), EA, GET_WIDTH)
-      => #wrap(GET_WIDTH, VALUE)
-      requires         GET_WIDTH <=Int SET_WIDTH
+       andBool 8 <=Int SHIFT
+       andBool #isByteMap(BM)
       [simplification]
 
-    rule #getRange(#setRange(BM, EA, VALUE, SET_WIDTH), EA, GET_WIDTH)
-      => #wrap(SET_WIDTH, VALUE)
-      requires (notBool GET_WIDTH <=Int SET_WIDTH)
-       andBool #getRange(BM, EA +Int SET_WIDTH, GET_WIDTH -Int SET_WIDTH) ==Int 0
```

Comment:
I take it these `#getRange(#setRange(...))` lemmas are no longer needed? Were any other removed?

Reply:
(As opposed to just moved)